### DISCUSSION TITLE
description text - add line feed?

### DISCUSSION THREAD

**Discussion by oldfart101** [Lissy93/dashy#412] [General]
as the title says is there a way to insert a CR or LF in the description text

**Comment by lissy93** [comment]
That's not possible, sorry.

**Reply by lissy93** [reply]
In [`Item.vue`](https://github.com/Lissy93/dashy/blob/master/src/components/LinkItems/Item.vue#L17), if you changed:
```html
<p class="description">{{ description }}</p>
```

to:
```html
<p class="description" v-html="description" />
```

Then this would allow you to render any HTML characters here, including `<br>` for new lines. But would also need to be taken care of to prevent introduction of an XSS vulnerability. So that would be up to you.

**Reply by oldfart101** [reply]
thanks for the reply
?? XSS vulnerability ??

**Reply by lissy93** [reply]
[XXS](https://owasp.org/www-community/attacks/xss/) stands for Cross-Site Scripting, and it's when malicious scripts are injected into trusted websites.

So in this example, if we used v-html here, the page would render whatever was passed into the description tag. Attackers can use this to render whatever they like, including malicious code (usually by injecting a `<script>` tag).

There is a way around this, and that it to sanitize the users input, say strip out all tags except for a list of pre-defined allowed tags (maybe like `<br>`, `<p>`, `<b>`, `<i>`, etc), but that adds a lot of complexity for a simple task, and complexity also makes testing harder.

I will have a think about whether it's worth trying to implement this, and will update you here if anything changes :)
